fastmri_recon/training_scripts/model_saving_workaround.py

```diff
@@ -1,9 +1,30 @@
+import os
+import pickle as pkl
+
 from tensorflow.keras.callbacks import ModelCheckpoint
+import tensorflow.keras.backend as K
+
 
 # taken from https://github.com/tensorflow/tensorflow/issues/39679
 class ModelCheckpointWorkAround(ModelCheckpoint):
+    def __init__(self, filepath, save_optimizer=True, **kwargs):
+        os.makedirs(os.path.dirname(filepath), exist_ok=True)
+        self.save_optimizer = save_optimizer
+        super(ModelCheckpointWithOptimizer, self).__init__(filepath=filepath, **kwargs)
```

Comment:
Watch out the naming

Reply:
Also why do you need the filepath logic?

Reply:
Well, I had codes failing just cause we didnt have the directory present :\. This is a good backup, anyway it just creates the output directory if needed, and if it exists, would not do anything.

Reply:
```suggestion
        super(ModelCheckpointWorkAround, self).__init__(filepath=filepath, **kwargs)
```